receiver/snmpreceiver/config.go

```diff
@@ -562,10 +579,21 @@ func validateResourceAttributeConfigs(cfg *Config) error {
 		return nil
 	}
 
-	// Make sure each Resource Attribute has either an OID or IndexedValuePrefix
+	// Make sure each Resource Attribute has either an OID or ScalarOID or IndexedValuePrefix
 	for attrName, attrCfg := range resourceAttributes {
-		if attrCfg.OID == "" && attrCfg.IndexedValuePrefix == "" {
-			combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgResourceAttributeNoOIDOrPrefix, attrName))
+		if attrCfg.OID == "" && attrCfg.ScalarOID == "" && attrCfg.IndexedValuePrefix == "" {
```

Comment:
As noted elsewhere, I think this function should probably handle all enforcement of whether or not these values are specified. Currently, this isn't actually validating that only one of oid, scalar_oid, indexed_value_prefix is set.

Reply:
It might make sense to convert all this validation to a switch statement. Roughly:
```go
hasOID := attrCfg.OID == ""
hasScalarOID := attrCfg.ScalarOID == ""
hasIVP := attrCfg.IndexedValuePrefix == ""

switch {
    case hasOID:
        if hasScalarOID || hasIVP { // err }
        // validate OID
    case hasScalarOID:
        if hasOID || hasIVP { // err }
        // validate Scalar OID
    case hasIVP:
        if hasOID || hasScalarOID { // err }
       // validate IVP (if any)
    default:
        // err must specify one
    }
}
```